src/replication.c

```diff
@@ -3476,6 +3494,23 @@ int replicationCountAcksByOffset(long long offset) {
     return count;
 }
 
+/* Return the number of replicas that already acknowledged the specified
+ * replication offset being AOF fsynced. */
+int replicationCountAOFAcksByOffset(long long offset) {
```

Comment:
Turns out it's possible to write a module and a Lua script that propagate to the AOF and doesn't propagate to the replication stream. see `REDISMODULE_ARGV_NO_REPLICAS` and `luaRedisSetReplCommand`.

That can result in two bad cases, scenario:
User executes command that only propagates to AOF, and then immediately issues a WAITAOF, and there's no further writes on the replication stream after that.
1. if the the last thing that happened on the replication stream is a PING (which increased the replication offset but won't trigger an fsync on the replica), then the client would hang forever (will wait for an fack that the replica will never send sine it doesn't trigger any fsyncs).
2. if the last thing that happened is a write command that got propagated properly, then WAITAOF will be released immediately, without waiting for an fsync (since the offset didn't change)

We need to decide if we want to handle that in some way, or ignore it as an abusive case (i'm not sure what's the justification for someone to explicitly propagate only to AOF).

On the other hand, if the client asked WAITAOF to wait for remote AOF files, and propagated a command only to the local AOF, maybe it deserves to get blocked forever (even if there are further writes), since his write didn't go to replicas and he waits for something that wouldn't happen..

So the other scenario we should look at is a case where there's no replication, only a master with AOF, and a command that's explicitly only propagating to AOF, and then user uses WAITAOF to wait for the local AOF file.
in that case, with the current code, we won't increment the replication offset and the client will get released before the fsync. that's probably easy to solve (increment the replication offset in that scenario).